src/services/impl/FileLogger.hh

```diff
@@ -29,6 +29,12 @@ public:
             pFileSystem.DeleteFile(sOldLogFilePath);
             pFileSystem.MoveFile(sLogFilePath, sOldLogFilePath);
         }
+        else if (nLogSize < 0)
+        {
+            std::wstring sCacheDirectory = pFileSystem.BaseDirectory() + L"RACache";
```

Comment:
I spoke too soon. The `+` should be replaced with a `StringPrintf` to only work on one string instead of needing two to make one.